src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -86,4 +109,121 @@ public function update(OrderReturn $orderReturn): void
             OrderReturnException::class
         );
     }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     *
+     * @throws CoreException
+     * @throws DeleteOrderReturnProductException
+     * @throws OrderReturnException
+     */
+    public function deleteOrderReturnProduct(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId
+    ): void {
+        $orderReturn = $this->get($orderReturnId);
+
+        if ((int) ($orderReturn->countProduct()) <= 1) {
+            throw new DeleteOrderReturnProductException(
+                'Can\'t delete last product from merchandise return',
+                DeleteOrderReturnProductException::LAST_ORDER_RETURN_PRODUCT
+            );
+        }
+        $orderReturnDetail = $this->getOrderReturnDetailByOrderDetailId($orderReturnDetailId->getValue());
+
+        if (!$orderReturnDetail) {
+            throw new DeleteOrderReturnProductException(
+                'Couldn\'t find merchandise return detail',
+                DeleteOrderReturnProductException::ORDER_RETURN_PRODUCT_NOT_FOUND
+            );
+        }
+
+        $this->deleteOrderReturnDetail(
+            $orderReturnDetail->getOrderReturnId(),
+            $orderReturnDetailId,
+            $orderReturnDetail->getCustomizationId()
+        );
+    }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     * @param CustomizationId|null $customizationId
+     *
+     * @throws DeleteOrderReturnProductException
+     */
+    public function deleteOrderReturnDetail(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId,
+        ?CustomizationId $customizationId = null
+    ): void {
+        try {
+            if (!OrderReturn::deleteOrderReturnDetail(
+                $orderReturnId->getValue(),
+                $orderReturnDetailId->getValue(),
+                $customizationId ? $customizationId->getValue() : 0
+            )) {
+                throw new DeleteOrderReturnProductException(
+                    'Failed to delete merchandise return detail',
+                    DeleteOrderReturnProductException::UNEXPECTED_ERROR
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new DeleteOrderReturnProductException(
+                'Failed to delete merchandise return detail',
+                DeleteOrderReturnProductException::UNEXPECTED_ERROR
+            );
+        }
+    }
+
+    public function getOrderReturnDetailByOrderDetailId(int $orderDetailId): ?OrderReturnDetail
```

Comment:
You can throw an exception if not found. That'll make the development easier as we will not need to check null vars.